### PR TITLE
Bls12 377 gadgets member naming (depends on #21)

### PR DESCRIPTION
**Pull request by dtebbs**
Change the convention for gadget member names and annotations.

### CODE REVIEW

**Comment by dtebbs** [comment]
#20 is addressed as part of these changes.

File: libzecale/circuits/fields/fp12_2over3over2_gadgets.hpp [hunk 1]
```diff
@@ -193,9 +193,9 @@ class Fp12_2over3over2_mul_gadget
     Fp12_2over3over2_variable<Fp12T> _A;
     Fp12_2over3over2_variable<Fp12T> _B;
     Fp12_2over3over2_variable<Fp12T> _result;
-    Fp6_3over2_mul_gadget<Fp6T> _v0;
-    Fp6_3over2_mul_gadget<Fp6T> _v1;
-    Fp6_3over2_mul_gadget<Fp6T> _a0_plus_a1_times_b0_plus_b1;
+    Fp6_3over2_mul_gadget<Fp6T> _compute_v0;
+    Fp6_3over2_mul_gadget<Fp6T> _compute_v1;
+    Fp6_3over2_mul_gadget<Fp6T> _compute_a0_plus_a1_times_b0_plus_b1;
```
Comment (AntoineRondelet) [inline libzecale/circuits/fields/fp12_2over3over2_gadgets.hpp:198]: A few lines above you have `_compute_z1z3z5_times_x0x2x4` and here you have `_compute_a0_plus_a1_times_b0_plus_b1`, which is not consistent

File: libzecale/circuits/fields/fp12_2over3over2_gadgets.tcc [hunk 2]
```diff
@@ -197,110 +199,110 @@ Fp12_2over3over2_mul_by_024_gadget<Fp12T>::Fp12_2over3over2_mul_by_024_gadget(
...
     // z0*x4 + z4*x0 = (z0 + z4)*(x0 + x4) - z0*x0 - z4*x4
     // out_z4 = z0*x4 + z4*x0 + non_residue * z5*x2
     //        = (z0 + z4)*(x0 + x4) - z0*x0 - z4*x4 + non_residue * z5*x2
     // => (z0 + z4)*(x0 + x4) = out_z4 + z0*x0 + z4*x4 - non_residue * z5*x2
-    , _z5_x2(
+    , _compute_z5_x2(
           pb,
           Z._c1._c2,
           X_2,
           libsnark::Fp2_variable<Fp2T>(pb, FMT(annotation_prefix, " z5*x2")),
```
Comment (AntoineRondelet) [inline libzecale/circuits/fields/fp12_2over3over2_gadgets.tcc:285]: If the gadget is `compute_z5_x2` (or `compute_z5_times_x2`), I'd call the output wire `z5_x2` (or `z5_times_x2`) then

File: libzecale/circuits/fields/fp12_2over3over2_gadgets.tcc [hunk 3]
```diff
@@ -129,22 +129,24 @@ Fp12_2over3over2_square_gadget<Fp12T>::Fp12_2over3over2_square_gadget(
...
           _A._c1,
           _result._c1 * (FieldT("2").inverse()),
-          FMT(annotation_prefix, " _alpha"))
-    , _beta(
+          FMT(annotation_prefix, " _compute_alpha"))
+    , _compute_beta(
           pb,
           _A._c0 + _A._c1,
           _A._c0 + fp6_mul_by_non_residue<Fp12T>(
                        pb, _A._c1, FMT(annotation_prefix, " a1*v")),
```
Comment (AntoineRondelet) [inline libzecale/circuits/fields/fp12_2over3over2_gadgets.tcc:142]: `a1_times_v` ?

File: libzecale/circuits/fields/fp12_2over3over2_gadgets.tcc [hunk 4]
```diff
@@ -129,22 +129,24 @@ Fp12_2over3over2_square_gadget<Fp12T>::Fp12_2over3over2_square_gadget(
...
           _result._c0 +
               fp6_mul_by_non_residue<Fp12T>(
-                  pb, _alpha._result, FMT(annotation_prefix, " alpha*v")) +
-              _alpha._result,
-          FMT(annotation_prefix, " _beta"))
+                  pb,
+                  _compute_alpha._result,
+                  FMT(annotation_prefix, " alpha*v")) +
```
Comment (AntoineRondelet) [inline libzecale/circuits/fields/fp12_2over3over2_gadgets.tcc:147]: `alpha_times_v`?

File: libzecale/circuits/fields/fp12_2over3over2_gadgets.hpp [hunk 5]
```diff
@@ -138,32 +138,32 @@ class Fp12_2over3over2_mul_by_024_gadget
...
-    libsnark::Fp2_mul_gadget<Fp2T> _z3_x0;
-    libsnark::Fp2_mul_gadget<Fp2T> _z24_x24;
+    libsnark::Fp2_mul_gadget<Fp2T> _compute_z3_x0;
+    libsnark::Fp2_mul_gadget<Fp2T> _compute_z24_x24;
 
     // out_z4 = z0*x4 + z4*x0 + non_residue * z5*x2
-    libsnark::Fp2_mul_gadget<Fp2T> _z5_x2;
-    libsnark::Fp2_mul_gadget<Fp2T> _z04_x04;
+    libsnark::Fp2_mul_gadget<Fp2T> _compute_z5_x2;
+    libsnark::Fp2_mul_gadget<Fp2T> _compute_z04_x04;
```
[the remark this answers is not in the document]
Comment (dtebbs) [reply libzecale/circuits/fields/fp12_2over3over2_gadgets.hpp:160]: I'm not sure I follow.  Which would be called `compute_X_times_Y`?

Following the implementation in libff, we use x0, x2, x4 and z0, ..., z5 and multiply everything in the same order.

[the remark this answers is not in the document]
Comment (AntoineRondelet) [reply libzecale/circuits/fields/fp12_2over3over2_gadgets.hpp:160]: Sorry that's not clear...
I mean rename `_compute_z5_x2` -> `_compute_z5_times_x2`, `_compute_z1_x2` ->` _compute_z1_times_x2` etc etc

File: libzecale/circuits/fields/fp12_2over3over2_gadgets.hpp (hunk 1 above)
Comment (dtebbs) [reply libzecale/circuits/fields/fp12_2over3over2_gadgets.hpp:199]: renamed `_compute_z1z3z5_times_x0x2x4`.

File: libzecale/circuits/fields/fp12_2over3over2_gadgets.tcc (hunk 2 above)
Comment (dtebbs) [reply libzecale/circuits/fields/fp12_2over3over2_gadgets.tcc]: changed all of these

File: libzecale/circuits/fields/fp12_2over3over2_gadgets.hpp (hunk 1 above)
Comment (AntoineRondelet) [reply libzecale/circuits/fields/fp12_2over3over2_gadgets.hpp:199]: Thanks.
To be clear, `_compute_z1z3z5_times_x0x2x4` was fine by me, i.e. you could also have decided to rename `_compute_a0_plus_a1_times_b0_plus_b1` as `_compute_a0a1_times_b0b1` to be consistent.

File: libzecale/circuits/fields/fp12_2over3over2_gadgets.tcc (hunk 3 above)
Comment (dtebbs) [reply libzecale/circuits/fields/fp12_2over3over2_gadgets.tcc]: changed annotation

File: libzecale/circuits/fields/fp12_2over3over2_gadgets.tcc (hunk 4 above)
Comment (dtebbs) [reply libzecale/circuits/fields/fp12_2over3over2_gadgets.tcc]: changed annotation

**Comment by AntoineRondelet** [comment]
LGTM

**Pull request merged**